sdk/PrebidServerAdapter/PBServerAdapter.m

```diff
@@ -103,121 +114,110 @@ - (NSURLRequest *)buildRequestForAdUnits:(NSArray<PBAdUnit *> *)adUnits {
     }
 }
 
-- (NSDictionary *)requestBodyForAdUnits:(NSArray<PBAdUnit *> *)adUnits {
+- (NSDictionary *)openRTBRequestBodyForAdUnits:(NSArray<PBAdUnit *> *)adUnits {
     NSMutableDictionary *requestDict = [[NSMutableDictionary alloc] init];
 
-    if (self.primaryAdServer == PBPrimaryAdServerMoPub || self.primaryAdServer == PBPrimaryAdServerUnknown) {
-        requestDict[@"cache_markup"] = @(1);
-    }
+    requestDict[@"id"] = @"some-request-id";
+    requestDict[@"test"] = @(1);
+    requestDict[@"tmax"] = @(500);
 
-    requestDict[@"sort_bids"] = @(1);
-    // need this so DFP targeting keys aren't too long
-    requestDict[@"max_key_length"] = @(20);
-    requestDict[@"account_id"] = self.accountId;
-    requestDict[@"tid"] = [[NSUUID UUID] UUIDString];
-    requestDict[@"prebid_version"] = @"0.21.0-pre";
+    requestDict[@"app"] = [self openrtbApp];
+    requestDict[@"device"] = [self openrtbDevice];
+    requestDict[@"user"] = [self openrtbUser];
+    requestDict[@"geo"] = [self openrtbGeo];
+    requestDict[@"imp"] = [self openrtbImpsFromAdUnits:adUnits];
+    requestDict[@"ext"] = [self openrtbRequestExtension];
 
-    requestDict[@"sdk"] = @{@"source": @"prebid-mobile",
-                            @"version": kPrebidMobileVersion,
-                            @"platform": @"iOS"};
+#ifdef DEBUG
+	requestDict[@"test"] = @(YES);
+#endif
 
-    NSDictionary *user = [self user];
-    if (user) {
-        requestDict[@"user"] = user;
-    }
-    NSDictionary *device = [self device];
-    if (device) {
-        requestDict[@"device"] = device;
-    }
-    NSDictionary *appID = [self app];
-    if (appID != nil) {
-        requestDict[@"app"] = appID;
-    }
-    NSArray *keywords = [self keywords];
-    if (keywords) {
-        requestDict[@"keywords"] = keywords;
+    return [requestDict copy];
+}
+
+- (NSDictionary *)openrtbRequestExtension {
+    NSMutableDictionary *requestPrebidExt = [[NSMutableDictionary alloc] init];
+    
+    if (self.shouldCacheLocal == FALSE) {
+        requestPrebidExt[@"cache"] = @{@"bids" : [[NSMutableDictionary alloc] init]};
     }
+    requestPrebidExt[@"targeting"] = @{@"lengthmax" : @(20)};
+    NSMutableDictionary *requestExt = [[NSMutableDictionary alloc] init];
```

Comment:
Add ("pricegranularity", "medium") for now before we can set storedrequest on the server.

Reply:
added now